### ISSUE TITLE
Spark Streaming Support

### ISSUE THREAD

**Issue by lucienfregosi** [AbsaOSS/spline-spark-agent#41]
Is it planned to add an integration with Spark Streaming ? It could be useful to be able to apply some lineage for batch and streaming data

**Comment by NickDudu** [comment]
Hi @wajda , May I confirm that the Structured Streaming is not supported such as writeStream API? Thanks

**Comment by wajda** [comment]
No, streaming is not supported due to fundamental problems with the definition and representation of data lineage in context of streaming. The topic remains unclear.

**Comment by NickDudu** [comment]
Hi @wajda No problem, thanks for the confirmation.

**Comment by jozefbakus** [comment]
Hello Everyone, 
We have been investigating spline and spark structured streaming.
We have been able to implement spline-agent for spark structured streaming using spark’s StreamingQueryListener, in a similar way as is described here: https://absaoss.github.io/spline/2018/10/04/Spline-Data-Lineage-For-Structured-Streaming.html (9:02 - 11:23). Code for our POC can be found here: https://github.com/jozefbakus/spline-spark-agent/pull/1 

Along the way we came across one major problem, linking. Linking in terms of connecting streaming parent-child lineages.
Currently, time linking is used: https://absaoss.github.io/spline/2018/10/04/Spline-Data-Lineage-For-Structured-Streaming.html (18:33 - 20:14). Time linking is not sufficient for streaming jobs. 
We are trying to find a suitable type of linking for streaming jobs. One of the solutions might be using kafka offsets similar way as described here: https://absaoss.github.io/spline/2018/10/04/Spline-Data-Lineage-For-Structured-Streaming.html (20:14 - 22:15).

To be able to link parent-child lineages, source and destination offsets (read and write offsets) are required. Spark gives us source offsets out of the box, the problem lies in destination offsets. Spark does not provide information about what offsets data was written to. Getting destination offsets in a nice, pluggable way is our current issue that we are trying to resolve before we can move forward.

Using read/write offsets linking might not be the only way, so we are also investigating different types of lineage linking.

**Comment by wajda** [comment]
The Spark Streaming support has been deprioritized, so I'm removing this feature from the active backlog.
